### PR TITLE
Hookup tiled mesh layer data provider creation, add empty shell for a future Cesium tiles provider

### PR DESCRIPTION
**Pull request by nyalldawson**
Another non-functional building block for future tiled mesh data providers.

(The next PR will be where the interesting stuff starts, because that's when I plan on refactoring a bunch of the existing layer/provider handling code to make it easier to introduce new layer types without introducing a ton of code duplication...)

### CODE REVIEW

File: src/core/tiledmesh/qgscesiumtilesdataprovider.cpp [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+#include "qgsapplication.h"
+#include "qgsprovidersublayerdetails.h"
+#include "qgsthreadingutils.h"
+
+#include <QUrl>
+#include <QIcon>
+
+///@cond PRIVATE
+
+#define PROVIDER_KEY QStringLiteral( "cesium" )
```
Comment (nirvn) [inline src/core/tiledmesh/qgscesiumtilesdataprovider.cpp:29]: In the off chance that cesium pushes another dataset format we'd want to embrace in the future, should we call this cesiummesh, cesiumtiles, or cesium3d instead?

File: src/core/tiledmesh/qgscesiumtilesdataprovider.cpp [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+//
+
+QgsCesiumTilesProviderMetadata::QgsCesiumTilesProviderMetadata():
+  QgsProviderMetadata( PROVIDER_KEY, PROVIDER_DESCRIPTION )
+{
+}
+
+QIcon QgsCesiumTilesProviderMetadata::icon() const
+{
+  return QgsApplication::getThemeIcon( QStringLiteral( "xxxcesiumxxx.svg" ) );
```
Comment (nirvn) [inline src/core/tiledmesh/qgscesiumtilesdataprovider.cpp:89]: Actual icon coming in a separate PR?

**Review by nirvn — APPROVED** [review]

File: src/core/tiledmesh/qgscesiumtilesdataprovider.cpp (hunk 2 above)
Comment (nyalldawson) [reply src/core/tiledmesh/qgscesiumtilesdataprovider.cpp:89]: Correct :wink: .. my current attempts at making a nice looking 3d tiles icon at the required small size are failing badly

File: src/core/tiledmesh/qgscesiumtilesdataprovider.cpp (hunk 1 above)
Comment (nyalldawson) [reply src/core/tiledmesh/qgscesiumtilesdataprovider.cpp:29]: I'll go with "cesiumtiles" :+1:

**Pull request merged**